x-pack/solutions/observability/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/badge/span_links_badge.tsx

```diff
@@ -45,13 +45,17 @@ export function SpanLinksBadge({ linkedParents, linkedChildren, id, onClick }: P
     >
       <EuiBadge
         data-test-subj={`spanLinksBadge_${id}`}
-        onClick={(e: any) => {
-          e.stopPropagation();
-          onClick(spanLinksFlyoutTab);
-        }}
-        onClickAriaLabel={i18n.translate('xpack.apm.waterfall.spanLinks.badgeAriaLabel', {
-          defaultMessage: 'Open span links details',
-        })}
+        {...(onClick
+          ? {
+              onClick: (e: any) => {
+                e.stopPropagation();
+                onClick(spanLinksFlyoutTab);
+              },
+              onClickAriaLabel: i18n.translate('xpack.apm.waterfall.spanLinks.badgeAriaLabel', {
+                defaultMessage: 'Open span links details',
+              }),
+            }
+          : [])}
```

Comment:
Do we want to spread an empty array? Perhaps just `undefined` here is fine

Reply:
<img width="1096" alt="Screenshot 2025-03-27 at 09 00 09" src="https://github.com/user-attachments/assets/db7d031c-7858-401c-b886-cc2ebd9930ba" />